### ISSUE TITLE
Message shown when sender sets "Only send to verified users" is misleading

### ISSUE THREAD

**Issue by kegsay** [vector-im/element-meta#2450]
Element-Web, Android (classic) and iOS (classic) all have the ability to enable a security flag which says something like:

>  In encrypted rooms, only send messages to verified users

(previously:  "never send encrypted messages to unverified devices": see https://github.com/element-hq/element-web/issues/29549)

However, if the recipient is not verified, they will see a misleading error: either a UTD (in the case of [Element iOS](https://github.com/element-hq/element-ios/issues/7825)) or "The sender has blocked you from receiving this message because *your device is unverified*" (which is incorrect, in the case of EW and EX).

 - [x] https://github.com/element-hq/element-web/issues/27653
 - [x] https://github.com/matrix-org/matrix-rust-sdk/issues/4270
 - [x] https://github.com/element-hq/element-ios/issues/7825
 - [x] https://github.com/element-hq/element-web/issues/28465
 - [ ] Element X

See also https://github.com/element-hq/element-meta/issues/2621

**Comment by BillCarsonFr** [comment]
This is actually a expected UTD, we should probably report it as such.
Check that we properly display the withheld code to the user, if not it's bug.

**Comment by richvdh** [comment]
Can confirm this doesn't work correctly in Web: https://github.com/element-hq/element-web/issues/27653

**Comment by richvdh** [comment]
https://github.com/element-hq/element-web/issues/27653 is now fixed, and EW now shows "The sender has blocked you from receiving this message because your device is unverified".

However, this remains a problem in Element iOS (https://github.com/element-hq/element-ios/issues/7825) and Element X.

**Comment by richvdh** [comment]
> https://github.com/element-hq/element-web/issues/27653 is now fixed, and EW now shows "The sender has blocked you from receiving this message because your device is unverified".

This isn't really the right error message, though. The "Never send encrypted messages to unverified sessions from this session" button isn't very clear what it means, but AFAICT it means that we won't send messages to verified devices belonging to *unverified* users (as well as unverified devices belonging to verified users).

In other words, the error message should be something like:

"The sender has blocked you from receiving this message because your device is unverified, or because they have not verified you". Which is obviously terrible, but at least not actively misleading.

**Comment by andybalaam** [comment]
Suggested wording:

"The sender's security settings prevented you receiving this message. This may be because your device is insecure, or they require you to perform user verification. Secure your device by verifying it, or verify with this user to fix this problem."

**Comment by richvdh** [comment]
I have opened https://github.com/element-hq/element-web/issues/28465 to track updating the wording in EW, and https://github.com/element-hq/element-meta/issues/2621 for the longer-term task of distinguishing between the two situations.

**Comment by richvdh** [comment]
> "The sender's security settings prevented you receiving this message. This may be because your device is insecure, or they require you to perform user verification. Secure your device by verifying it, or verify with this user to fix this problem."

Great, but maybe too wordy, particularly for a mobile client? @americanrefugee I wonder if you can help us here?

**Comment by americanrefugee** [comment]
I'm having a wee bit of trouble fully understanding the use case, but I would say...

Scenario 1: If the user has an unverified device but the sender does NOT require user verification:

- "You need to verify your device to see this message. <Verify device>"

Scenario 2: If the user has a verified device but the sender requires user verification:

- "You need to verify (username) to see messages from them. <Go to profile>"

Scenario 3: If the user has an unverified device AND the sender requires user verification.

- "You need to verify your device and then verify (username) to see messages from them. <Verify device>"

**Comment by americanrefugee** [comment]
How about this instead:

1st case: The recipient's device is not secure, and we don't know if the sender selected the security option

- "You need to verify your device and may need to verify the sender's identity to see messages from them. Verify this device (text link)"

2nd case: The recipients device is secure, but they still can't see the message(s)

- "You need to verify the sender's identity to see messages from them. Go to their profile (text link)"

**Comment by americanrefugee** [comment]
Final wording:

- "This message was blocked either because your device is unverified or because the sender needs to verify your identity."

**Comment by BillCarsonFr** [comment]
> Final wording:
> 
>     * "This message was blocked either because your device is unverified or because the sender needs to verify your identity."

@americanrefugee The [crypto terminology talks about ](https://github.com/matrix-org/matrix-spec-proposals/blob/andybalaam/crypto-terminology/proposals/4161-crypto-terminology.md#devices)`Insecure Device` but not `unverified` device.

So maybe we should say:
- "This message was blocked either because your device is insecure or because...`
or
- "This message was blocked either because your did not verify your device  or because...`

**Comment by americanrefugee** [comment]
> @americanrefugee The [crypto terminology talks about ](https://github.com/matrix-org/matrix-spec-proposals/blob/andybalaam/crypto-terminology/proposals/4161-crypto-terminology.md#devices)`Insecure Device` but not `unverified` device.
> 
> So maybe we should say:
> 
> * "This message was blocked either because your device is insecure or because...`
>   or
> * "This message was blocked either because your did not verify your device or because...`

I'd go with the second option. We ask the user to "verify" their device during sign in, not "secure" their device. So I think we at least need to use the use terminology everywhere.

**Comment by BillCarsonFr** [comment]
https://localazy.com/p/element/source-language/_a7139002366977292897
`timeline.decryption_failure.withheld.unverified`

**Comment by andybalaam** [comment]
IRL discussion 2025-03-24 with @mxandreas - making this message more ambiguous doesn't make things much better.

Instead, we should do https://github.com/element-hq/element-web/issues/29549 to improve the label on the toggle that causes this problem, and then later do https://github.com/element-hq/element-meta/issues/2621 to distinguish these different situations and provide different messages.

**Comment by richvdh** [comment]
keeping this open until https://github.com/element-hq/element-meta/issues/2621 is fixed
